torch/_dynamo/symbolic_convert.py

```diff
@@ -329,24 +329,30 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
                 self.jump(inst)
         elif isinstance(value, UserDefinedObjectVariable):
             x = value.var_getattr(self, "__bool__")
-            # __bool__ is function
+            # if __bool__ is missing, trying __len__ to infer a truth value.
+            if x.is_python_constant() and x.as_python_constant() is None:
+                x = value.var_getattr(self, "__len__")
+            # __bool__ or __len__ is function
             if isinstance(x, UserMethodVariable):
                 state = self.copy_graphstate()
                 result = x.call_function(self, [], {})
-                if isinstance(result, ConstantVariable) and isinstance(
-                    result.value, bool
+                if isinstance(result, ConstantVariable) and (
+                    x.fn.__name__ == "__bool__"
+                    and isinstance(result.value, bool)
+                    or x.fn.__name__ == "__len__"
+                    and isinstance(result.value, int)
```

Comment:
This seems too restrictive.  I think this can just be `isinstance(result.value, (bool, int))`.  Or maybe just skipped entirely.